src/components/dashboard/EditClient.tsx

```diff
@@ -49,11 +52,29 @@ const Profile = () => {
       onCompleted={onCompleted}
       top={STICKY_BAR_HEIGHT + CONTEXT_HEADER_HEIGHT}
       title={
-        <>
-          <Typography variant='h3' sx={{ pt: 0, pb: 4 }}>
+        <Stack direction='row' justifyContent='space-between'>
+          <Typography variant='h3' sx={{ pt: 0, pb: 4 }} flexGrow={1}>
             Edit Client Details
           </Typography>
-        </>
+          {client.id && (
+            <div>
+              <RootPermissionsFilter permissions='canDeleteClients'>
+                <DeleteClientButton
+                  clientId={client.id}
+                  variant='text'
+                  onCompleted={(data) => {
+                    if (
+                      isEmpty(data.deleteClient?.errors) &&
+                      data.deleteClient?.client
+                    ) {
+                      navigate(generateSafePath(Routes.CLIENT_SEARCH));
+                    }
+                  }}
```

Comment:
Nitpick, but I think this is a bit confusing, it made me thing we were ignoring the errors. I think it would be cleaner to just call the callback on successful deletion:
```suggestion
                  onSuccess={() => navigate(generateSafePath(Routes.CLIENT_SEARCH))}
```